core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -122,14 +122,16 @@ void clear() {
 
         @Override
         protected void store(ConfidentialKey key, byte[] payload) throws IOException {
-            LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getDigestOf(Util.toHexString(payload)));
+            //called only from tests, get hex string of sha 256 for logging payload
+            LOGGER.log(Level.FINER, "storing {0} {1}",  new Object[]{key.getId(), Util.getHexOfSHA256DigestOf(payload)});
             data.put(key.getId(), payload);
         }
 
         @Override
         protected byte[] load(ConfidentialKey key) throws IOException {
             byte[] payload = data.get(key.getId());
-            LOGGER.fine(() -> "loading " + key.getId() + " " + (payload != null ? Util.getDigestOf(Util.toHexString(payload)) : "null"));
+            //called only from tests, get hex string of sha 256 for logging payload
+            LOGGER.log(Level.FINER, "loading {0} {1}",  new Object[]{key.getId(), Util.getHexOfSHA256DigestOf(payload)});
```

Comment:
this obtains the digest of the payload even if it is not logged.

Reply:
Included condition to check if required log level is enabled before obtaining the digest